### PR TITLE
Tabulate EA/EH environment variables

### PR DESCRIPTION
**Pull request by richma-ms**
Tabulating all the known settings knobs via environment variables for official documentation.  I tried to track down everything, but the config readers are kinda spread out everywhere, so I might have missed a few.

I've also taken the liberty to mark some of these as omitted/obsolete if I don't think they should be customer facing.  Lemme know if you'd like any of those included, otherwise I'll have them removed from this doc before merging.

### CODE REVIEW

File: doc/EnvironmentVariables.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+| RuntimeLogLevel                      | Runtime diagnostic logging level, **fatal / error / warning / info / debug / verbose**, defaults to **info**             |
+| SendRuntimeQualityTelemetry          | Whether to enable sending runtime diagnostics metric, defaults to **true**                                               |
+| Storage_LogLevel                     | RocksDB diagnostic log level, **NONE / FATAL / HEADER / ERROR / WARN / INFO / DEBUG**, defaults to **NONE**              |
+| StorageClassName                     | StorageClassName to be used when creating a PersistedVolumeClaim                                                         |
+| StorageFolder                        | Path to place the EdgeHub database directory, defaults to **TempPath** of the current OS                                 |
+| UpstreamProtocol                     | Protocol used to for upstream connections, defaults to **Amqp**, and falls back to AmqpWs                                |
+| UseMountSourceForVolumeName          | ???, defaults to **false**                                                                                               |
```
Comment (richma-ms) [inline doc/EnvironmentVariables.md:31]: @darobs I couldn't figure out what this does, can you help me write a short description?

File: doc/EnvironmentVariables.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+## EdgeAgent
+| Variable                             | Description                                                                                                              |
```
Comment (varunpuranik) [inline doc/EnvironmentVariables.md:2]: Thanks for this. Can you pls add a third column for Default Value?

File: doc/EnvironmentVariables.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+| StorageClassName                     | StorageClassName to be used when creating a PersistedVolumeClaim                                                         |
+| StorageFolder                        | Path to place the EdgeHub database directory, defaults to **TempPath** of the current OS                                 |
+| UpstreamProtocol                     | Protocol used to for upstream connections, defaults to **Amqp**, and falls back to AmqpWs                                |
+| UseMountSourceForVolumeName          | ???, defaults to **false**                                                                                               |
+| UsePersistentStorage                 | Whether to save deployment config and module states to disk, defaults to **true**                                        |
+| UseServerHeartbeat                   | Sets the client-side heartbeat interval to 60sec for the Agent's upstream AMQP connection, defaults to **true**          |
+| EdgeDeviceHostName                   | omit?                                                                                                                    |
```
Comment (varunpuranik) [inline doc/EnvironmentVariables.md:34]: All of the ones below can be omitted.

File: doc/EnvironmentVariables.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
+## EdgeAgent
+| Variable                             | Description                                                                                                              |
+|--------------------------------------|--------------------------------------------------------------------------------------------------------------------------|
+| BackupConfigFilePath                 | Path to put the backup deployment config file                                                                            |
```
Comment (varunpuranik) [inline doc/EnvironmentVariables.md:4]: This is great - but can we split them into 2 categories - actual knobs that you can set, and knobs that customers should leave alone in most cases?

File: doc/EnvironmentVariables.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+| EDGEDEVICEHOSTNAME                      | omit?                                                                                                                         |
+| IOTEDGE_IOTHUBHOSTNAME                  | omit?                                                                                                                         |
+| IOTEDGE_DEVICEID                        | omit?                                                                                                                         |
+| IOTEDGE_MODULEID                        | omit?                                                                                                                         |
+| IotHubConnectionString                  | omit?                                                                                                                         |
+| usePersistentStorage                    | obsolete?                                                                                                                     |
```
Comment (varunpuranik) [inline doc/EnvironmentVariables.md:110]: This is needed (allows running EH with in memory storage)

File: doc/EnvironmentVariables.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+| storeAndForwardEnabled                  | obsolete?                                                                                                                     |
+| EnableNonPersistentStorageBackup        | obsolete?                                                                                                                     |
```
Comment (varunpuranik) [inline doc/EnvironmentVariables.md:112]: This is needed as well.

File: doc/EnvironmentVariables.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+| Storage_LogLevel                        | RocksDB diagnostic log level, **NONE / FATAL / HEADER / ERROR / WARN / INFO /DEBUG**, defaults to **NONE**                    |
+| StorageFolder                           | Path to place the EdgeHub database directory, defaults to **TempPath** of the current OS                                      |
+| UpstreamFanOutFactor                    | Max number of message groups to concurrently process for sending, grouped by sender, defaults to **10**                       |
+| UpstreamProtocol                        | Protocol used to for upstream connections, defaults to **Amqp**, and falls back to AmqpWs                                     |
+| UseServerHeartbeat                      | Sets the client-side heartbeat interval to 60sec for upstream AMQP connections, defaults to **true**                          |
+| experimentalFeatures                    | omit?                                                                                                                         |
```
Comment (varunpuranik) [inline doc/EnvironmentVariables.md:98]: Yes, omit all of these, except the 2 below.

**Review by varunpuranik — COMMENTED** [review]
Looks good, added some comments.

File: doc/EnvironmentVariables.md (hunk 1 above)
Comment (davebob3) [reply doc/EnvironmentVariables.md:31]: Something like this: "Tells the k8s conversion to use the volume mount source as persistent volume name." You can have it refer to the [values.yaml](https://github.com/Azure/iotedge/blob/master/kubernetes/charts/edge-kubernetes/values.yaml#L142) entry for it.

File: doc/EnvironmentVariables.md (hunk 3 above)
Comment (davebob3) [reply doc/EnvironmentVariables.md:34]: Do we want to omit them all or just state these are reserved by edge runtime?

**Review by davebob3 — COMMENTED** [review]
Always happy for more documentation.  All of the things that you have listed as "omit?" we may just want to say are common across all modules and are used by IoT Edge Runtime.

File: doc/EnvironmentVariables.md (hunk 5 above)
Comment (and-rewsmith) [reply doc/EnvironmentVariables.md:110]: This is duplicated above with a description. The difference is this one's first char isn't capitalized.

File: doc/EnvironmentVariables.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+| RuntimeLogLevel                      | Runtime diagnostic logging level                                                                         | fatal, error, warning, info, debug, verbose   | info                       |
+| SendRuntimeQualityTelemetry          | Whether to enable sending runtime diagnostics metric                                                     | bool                                          | true                       |
+| Storage_LogLevel                     | RocksDB diagnostic log level                                                                             | NONE, FATAL, HEADER, ERROR, WARN, INFO, DEBUG | NONE                       |
+| StorageClassName                     | StorageClassName to be used when creating a PersistedVolumeClaim                                         | string                                        |                            |
+| StorageFolder                        | Path to place the EdgeHub database directory, defaults to TempPath of the current OS                     | string                                        |                            |
```
Comment (varunpuranik) [inline doc/EnvironmentVariables.md:29]:
```suggestion
| StorageFolder                        | Path to place the EdgeAgent database directory, defaults to TempPath of the current OS                     | string                                        |                            |
```

**Review by varunpuranik — APPROVED** [review]

**Review by varunpuranik — APPROVED** [review]

**Pull request merged**